### PR TITLE
feat: add rfc3339 format for templates (fix #61)

### LINKED ISSUE #61: Please provide template variable for RFC3339 date format
Another tool I'm using to read these notes operates on dates with RFC3339 formatting.  It wold be handy to have such a template variable to use for this purpose.

### PR DESCRIPTION
**Pull request by lambtho12**
Add date format follwing rfc3339 for templates.

Fix #61

### CODE REVIEW

File: lua/telekasten.lua [hunk 1]
```diff
@@ -282,6 +282,14 @@ local function calculate_dates(date)
         .. ", "
         .. dinfo.year
 
+    local zonehour = string.sub(os.date("%z"), 1, 3)
+    local zonemin = string.sub(os.date("%z"), 4, 5)
+    dates.rfc3339 = os.date("%Y-%m-%dT%H:%M:%S")
```
Comment (renerocksai) [inline lua/telekasten.lua:287]: these don't take `dinfo` into account. So the `rfc3339` date will only work for the current time.

The function, however, is also used when selecting a date from the calendar. Please take this into account.

**Review by renerocksai — CHANGES_REQUESTED** [review]
I like the idea and it seems like a valuable contribution. Pls take the `dinfo` into consideration, though. Then it will be an easy merge.

**Comment by lambtho12** [comment]
I think this should solve it. 

As RFC3339 needs the hours, minutes and seconds it is unclear what should be used for other dates than today... This will always use the time of the note creation. Not sure if it should default to some other value or not.

**Review by renerocksai — APPROVED** [review]
That should do the trick! I don't mind a default time. One could argue to set it to the start of the day if not known - but anything's fine for me, really.

**Comment by renerocksai** [comment]
Sorry I just noticed your update now. Happy to merge, though!

**Pull request merged**